src/FsAutoComplete.Core/FileSystem.fs

```diff
@@ -214,7 +214,11 @@ type NamedText(fileName: string<LocalPath>, str: string) =
 
   /// Provides safe access to a substring of the file via FCS-provided Range
   member x.GetText(m: FSharp.Compiler.Text.Range) : Result<string, string> =
-    if not (Range.rangeContainsRange x.TotalRange m) then
+    // indexing into first line of empty file can be encountered when typing from an empty file
+    // if we don't check it, GetLineString will throw IndexOutOfRangeException
+    if (x :> ISourceText).GetLineCount() = 0 then
```

Comment:
We should probably add an `IsEmpty` onto the `IFSACSourceText` interface.

Reply:
Is the use of this `IsEmpty` widespread enough to make a change to public API?